android/cli/commands/_build.js

```diff
@@ -798,16 +798,19 @@ AndroidBuilder.prototype.config = function config(logger, config, cli) {
 											_t.keystoreAliases.pop();
 										}
 
+										// Parse the keystore's alias name and signature algorithm.
+										// Note: Algorithm can return "MD5withRSA (weak)" on JDK 8 and higher.
+										//       Only extract 1st token since we need a valid algorithm name.
 										const aliasRegExp = /Alias name: (.+)/,
-											sigalgRegExp = /Signature algorithm name: (.+)/;
+											sigalgRegExp = /Signature algorithm name: (.[^\s]+)/;
```

Comment:
that regex looks a little funky in that group. It went from saying: 'one or more of any character' to: 'one of any character followed by one or more non-spaces'. It probably works fine but may be a little misleading.

Reply:
Do you have a better solution? Your regex kung-fu is likely better than mine.
(I prefer to parse the old-school way via `indexOf()`.)

I tested my regex with the following script locally...
```javascript
var regex = /Signature algorithm name: (.[^\s]+)/;
//var text = "Signature algorithm name: MD5withRSA";
//var text = "Signature algorithm name:  MD5withRSA";
var text = "Signature algorithm name: MD5withRSA (weak)";
//var text = "Signature algorithm name: MD5withRSA  ( weak )  ";
//var text = "Signature algorithm name:  MD5withRSA  ( weak )  ";
//var text = "Signature algorithm name: SHA256withRSA";
//var text = "Signature algorithm name: ";
text = "The text above.\n" + text + "\nThe text below.\n";
var result = text.match(regex);
console.log(JSON.stringify(result));
```